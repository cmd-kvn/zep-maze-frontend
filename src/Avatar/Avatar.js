import React from 'react';
import  Emojify from 'react-emojione';
import { connect } from 'react-redux';
import styles from './Avatar.css'

export default function Avatar() {
    
    return ( <div className={styles.avatar}> <Emojify><span>:wink:</span></Emojify> </div>);
}
