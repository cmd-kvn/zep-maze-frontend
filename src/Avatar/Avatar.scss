

:local(.avatar){
    // margin-left: -16px;
    // margin-top: -16px;
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    padding: auto;
    display: flex;   
    justify-content: center;
    align-items: center;
}