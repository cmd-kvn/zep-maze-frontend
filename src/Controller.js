import React, { Component } from 'react';
import { goUP, goDOWN, goRIGHT, goLEFT } from './Controller.actions';
import { connect } from 'react-redux';

class Controller extends Component {
    
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        window.addEventListener('keypress', function(event) {
            if( event.keycode === 38 ) return goUP();
            if( event.keycode === 40 ) return goDOWN();
            if( event.keycode === 37 ) return goLEFT();
            if( event.keycode === 39 ) return goRIGHT();
        })
    }

    componentWillUnmount(){
        window.removeEventListener();
    }

    render() {
        return (
            <div className="Button-Bar">
                <button onClick={goUP}>Up</button>
                <button onClick={goDOWN}>Down</button>
                <button onClick={goLEFT}>Left</button>
                <button onClick={goRIGHT}>Right</button>
            </div>
        )
    }
}

export default connect (state => state)(Controller);