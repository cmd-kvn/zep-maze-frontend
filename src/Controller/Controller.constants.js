export const MOVE = 'MOVE';
export const GAME_WIN = 'GAME_WIN';
export const GAME_OVER = 'GAME_OVER';