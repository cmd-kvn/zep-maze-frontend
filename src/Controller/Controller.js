import React, { Component } from 'react';
// import { goUP, goDOWN, goRIGHT, goLEFT } from './Controller.actions';
import { connect } from 'react-redux';

class Controller extends Component {
    
    constructor(props) {
        super(props);
    }

    handleMove = ({keycode}) => {
        if( keycode === 38 ) return this.props.goUP();
        if( keycode === 40 ) return this.props.goDOWN();
        if( keycode === 37 ) return this.props.goLEFT();
        if( keycode === 39 ) return this.props.goRIGHT();
    }

    componentDidMount() {
        window.addEventListener('keypress', this.handleMove );
    }

    componentWillUnmount(){
        window.removeEventListener('keypress', this.handleMove );
    }

    render() {
        return (
            <div className="Button-Bar">
                <button onClick={this.props.goUP}>Up</button>
                <button onClick={this.props.goDOWN}>Down</button>
                <button onClick={this.props.goLEFT}>Left</button>
                <button onClick={this.props.goRIGHT}>Right</button>
            </div>
        )
    }
}

export default Controller;