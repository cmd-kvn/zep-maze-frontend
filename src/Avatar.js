import React from 'react';
import {emojify} from 'react-emojione';
import { UP, DOWN, LEFT, RIGHT } from './Controller.constants';

export function Avatar({ stance }) {
    switch(stance) {
        case UP: return ( <div> emojify( ':point_up_tone3:' ) </div> );
        case DOWN: return ( <div> emojify( ':point_down_tone3:' ) </div> );
        case RIGHT: return ( <div> emojify(':point_right_tone3:') </div> );
        case LEFT: return ( <div> emojify( ':point_left_tone3:' ) </div> );
        default: return ( <div> emojify( ':levitate_tone3:' ) </div>);
    }
}