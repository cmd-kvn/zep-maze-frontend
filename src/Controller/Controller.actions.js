import { UP, DOWN, LEFT, RIGHT } from './Controller.constants';


export function goUP() {
    return {
        type: UP,
        payload: null
    };
}

export function goDOWN() {
    return {
        type: DOWN,
        payload: null
    };
}

export function goRIGHT() {
    return {
        type: RIGHT,
        payload: null
    };
}

export function goLEFT() {
    return {
        type: LEFT,
        payload: null
    };
}