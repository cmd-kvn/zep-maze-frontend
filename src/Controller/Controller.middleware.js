import { MOVE, GAME_OVER, GAME_WIN } from './Controller.constants';
import { WALL, END } from '../Tile/Tile.constants';

export const moveChecker = store => next => action => {
  if(action.type !== MOVE ) return next(action);

  const { matrix: maze } = store.getStore();
  const [x, y] = action.payload;
  const tile = maze[y][x];  
  if(!tile || tile === WALL) return;
  if(tile === END) return next({ type: GAME_WIN });
  // if(tile === FIRE) return next({ type: GAME_OVER });
  
  // if(tile === DOOR) {
  //   const { inventory } = store.getState();
  //   const index = inventory.indexOf(KEY);
  //   if(index === -1) return;

  //   next({ type: USE_INVENTORY, payload: index });
  //   next({ type: CHANGE_TILE, payload: [x, y] });
  // }

  next(action);

}