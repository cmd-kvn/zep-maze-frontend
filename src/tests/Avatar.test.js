import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Avatar } from '../Avatar/Avatar';
import { UP, DOWN, LEFT, RIGHT } from '../Controller/Controller.constants';

describe('Avatar Component', () => {

    it('renders a levitating man emoji by default', () => {
        const wrapper = shallow(<Avatar />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders a left pointing emoji when given left stance', () => {
        const wrapper = shallow(<Avatar stance={LEFT}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders a right pointing emoji when given right stance', () => {
        const wrapper = shallow(<Avatar stance={RIGHT}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders a up pointing emoji when given up stance', () => {
        const wrapper = shallow(<Avatar stance={UP}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders a down pointing emoji when given down stance', () => {
        const wrapper = shallow(<Avatar stance={DOWN}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
})