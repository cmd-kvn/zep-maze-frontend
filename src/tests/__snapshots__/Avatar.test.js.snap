// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component renders a down pointing emoji when given down stance 1`] = `
<div
  className="Avatar"
>
   emojify( ':point_down_tone3:' ) 
</div>
`;

exports[`Avatar Component renders a left pointing emoji when given left stance 1`] = `
<div
  className="Avatar"
>
   emojify( ':point_left_tone3:' ) 
</div>
`;

exports[`Avatar Component renders a levitating man emoji by default 1`] = `
<div
  className="Avatar"
>
   emojify( ':levitate_tone3:' ) 
</div>
`;

exports[`Avatar Component renders a right pointing emoji when given right stance 1`] = `
<div
  className="Avatar"
>
   emojify(':point_right_tone3:') 
</div>
`;

exports[`Avatar Component renders a up pointing emoji when given up stance 1`] = `
<div
  className="Avatar"
>
   emojify( ':point_up_tone3:' ) 
</div>
`;
